tasks/upload.py

```diff
@@ -991,13 +991,15 @@ def possibly_setup_webhooks(self, commit, repository_service):
                     )
                     repository.hookid = hookid
                     repo_data["repo"]["hookid"] = hookid
+                    repository.webhook_secret = webhook_secret
```

Comment:
So we save the webhook if `should_post_webhook` or before?

Reply:
yes, only add the `webhook_secret` to the `repository` if the `create_webhook` or `webhook_update` update goes through without error